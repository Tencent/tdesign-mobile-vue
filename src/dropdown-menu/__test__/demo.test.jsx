/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import customizedMenuVue from '@/dropdown-menu/demos/customized-menu.vue';
import demoContainerVue from '@/dropdown-menu/demos/demo-container.vue';
import eventsVue from '@/dropdown-menu/demos/events.vue';
import mobileVue from '@/dropdown-menu/demos/mobile.vue';
import multiVue from '@/dropdown-menu/demos/multi.vue';
import optionsLayoutVue from '@/dropdown-menu/demos/options-layout.vue';
import singleVue from '@/dropdown-menu/demos/single.vue';

const mapper = {
  customizedMenuVue,
  demoContainerVue,
  eventsVue,
  mobileVue,
  multiVue,
  optionsLayoutVue,
  singleVue,
};

describe('DropdownMenu', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`DropdownMenu ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
