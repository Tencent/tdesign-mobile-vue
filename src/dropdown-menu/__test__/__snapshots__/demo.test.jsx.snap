// Vitest Snapshot v1

exports[`DropdownMenu > DropdownMenu customizedMenuVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown-menu"
>
  
  <div
    class="t-dropdown-menu"
  >
    
    <div
      class="t-dropdown-menu__item t-dropdown-menu__item--disabled"
    >
      <div
        class="t-dropdown-menu__title"
      >
        单选菜单
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        自定义内容
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    
    
    <!--v-if-->
    <!--v-if-->
    
  </div>
  <p>
     开关 1: 
    <strong>
      false
    </strong>
  </p>
  <p>
     开关 2: 
    <strong>
      true
    </strong>
  </p>
  <p>
     开关 3: 
    <strong>
      false
    </strong>
  </p>
  
</div>
`;

exports[`DropdownMenu > DropdownMenu demoContainerVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown-menu"
>
  
  
</div>
`;

exports[`DropdownMenu > DropdownMenu eventsVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown-menu"
>
  
  <div
    class="t-dropdown-menu"
  >
    
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        单选菜单
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        多选菜单
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    
    
    <!--v-if-->
    <!--v-if-->
    
  </div>
  <p>
     单选菜单 选中项: 
    <strong>
      option_2
    </strong>
  </p>
  <p>
     多选菜单 选中项: 
    <strong>
      options_A, options_C
    </strong>
  </p>
  
</div>
`;

exports[`DropdownMenu > DropdownMenu mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    DropdownMenu 下拉菜单
  </h1>
  <p
    class="summary"
  >
    菜单呈现数个并列的选项类目，用于整个页面的内容筛选，由菜单面板和菜单选项组成。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        单选下拉菜单
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-dropdown-menu"
      >
        
        <div
          class="t-dropdown-menu__item"
        >
          <div
            class="t-dropdown-menu__title"
          >
            全部产品
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <div
          class="t-dropdown-menu__item"
        >
          <div
            class="t-dropdown-menu__title"
          >
            默认排序
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        
        
        <!--v-if-->
        <!--v-if-->
        
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        分栏下拉菜单
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-dropdown-menu"
      >
        
        <div
          class="t-dropdown-menu__item"
        >
          <div
            class="t-dropdown-menu__title"
          >
            单列多选
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <div
          class="t-dropdown-menu__item"
        >
          <div
            class="t-dropdown-menu__title"
          >
            双列多选
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <div
          class="t-dropdown-menu__item"
        >
          <div
            class="t-dropdown-menu__title"
          >
            三列多选
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        
        
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
        
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        禁用状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-dropdown-menu"
      >
        
        <div
          class="t-dropdown-menu__item t-dropdown-menu__item--disabled"
        >
          <div
            class="t-dropdown-menu__title"
          >
            禁用菜单
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <div
          class="t-dropdown-menu__item t-dropdown-menu__item--disabled"
        >
          <div
            class="t-dropdown-menu__title"
          >
            禁用菜单
          </div>
          <svg
            class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M11 6H5l3 4.5L11 6z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        
        
        <!--v-if-->
        <!--v-if-->
        
      </div>
      
    </div>
  </div>
  <div
    style="height: 40px;"
  />
</div>
`;

exports[`DropdownMenu > DropdownMenu multiVue demo works fine 1`] = `
<div
  class="t-dropdown-menu"
>
  
  <div
    class="t-dropdown-menu__item"
  >
    <div
      class="t-dropdown-menu__title"
    >
      单列多选
    </div>
    <svg
      class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11 6H5l3 4.5L11 6z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <div
    class="t-dropdown-menu__item"
  >
    <div
      class="t-dropdown-menu__title"
    >
      双列多选
    </div>
    <svg
      class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11 6H5l3 4.5L11 6z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <div
    class="t-dropdown-menu__item"
  >
    <div
      class="t-dropdown-menu__title"
    >
      三列多选
    </div>
    <svg
      class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11 6H5l3 4.5L11 6z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  
  
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
</div>
`;

exports[`DropdownMenu > DropdownMenu optionsLayoutVue demo works fine 1`] = `
<div
  class="tdesign-demo-dropdown-menu"
  data-v-fde09476=""
>
  
  <div
    class="t-dropdown-menu"
    data-v-fde09476=""
  >
    
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        单列
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        双列
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    <div
      class="t-dropdown-menu__item"
    >
      <div
        class="t-dropdown-menu__title"
      >
        三列
      </div>
      <svg
        class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M11 6H5l3 4.5L11 6z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </div>
    
    
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    
  </div>
  <p
    data-v-fde09476=""
  >
     单列菜单 选中项: 
    <strong
      data-v-fde09476=""
    />
  </p>
  <p
    data-v-fde09476=""
  >
     两栏菜单 选中项: 
    <strong
      data-v-fde09476=""
    />
  </p>
  <p
    data-v-fde09476=""
  >
     三栏菜单 选中项: 
    <strong
      data-v-fde09476=""
    >
      options_A, options_C
    </strong>
  </p>
  
</div>
`;

exports[`DropdownMenu > DropdownMenu singleVue demo works fine 1`] = `
<div
  class="t-dropdown-menu"
>
  
  <div
    class="t-dropdown-menu__item"
  >
    <div
      class="t-dropdown-menu__title"
    >
      全部产品
    </div>
    <svg
      class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11 6H5l3 4.5L11 6z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <div
    class="t-dropdown-menu__item"
  >
    <div
      class="t-dropdown-menu__title"
    >
      默认排序
    </div>
    <svg
      class="t-icon t-icon-caret-down-small t-dropdown-menu__arrow"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M11 6H5l3 4.5L11 6z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  
  
  <!--v-if-->
  <!--v-if-->
  
</div>
`;
